samza-log4j/src/main/java/org/apache/samza/logging/log4j/StreamAppender.java

```diff
@@ -260,6 +275,11 @@ protected void setupSystem() {
 
     setSerde(log4jSystemConfig, systemName, streamName);
 
+    // Explicitly create stream appender stream with the partition count the same as the number of containers.
+    StreamSpec streamSpec = StreamSpec.createStreamAppenderStreamSpec(streamName, systemName, getPartitionCount());
```

Comment:
Log at info that we're creating the log stream x on system y with num partitions n.

Reply:
Sounds good will do.